docs/src/architecture/notes.md

```diff
@@ -1,73 +1,194 @@
+
+The `SyncState` API requires the Miden Client to provide a `note_tag` value which is used as a filter in the response. 
+
+## Note consumption
+As with creation, notes can only be consumed in Miden transactions. If a valid transaction consuming an `InputNote` gets verified by the Miden Node, the note's unique nullifier gets added to the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database) and is therefore consumed. Some notes can be consumed by any account and for some note consumption is restricted.
+
+### Note recipient to restrict note consumption to certain targets
+There are several ways to restrict the set of accounts that can consume a specific note. One way is to specifically define the target account ID as done in the P2ID and P2IDR note scripts. Another way is by using the concept of a `RECIPIENT`. Miden defines a `RECIPIENT` as: `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)` represented as `Word`. This concept restricts note consumption to those users who know the pre-image data of `RECIPIENT` - which might be a bigger set than a single account.   
+
+During the [transaction prologue](../transactions/transaction-kernel.md) the users needs to provide all the data to compute the note hash. That means, one can create notes that can only be consumed if the `serial_num` and other data is known. This information can be passed on off-chain by the sender to the recipient. 
+
+For public notes, anyone can compute the note hash and the concept of the `RECIPIENT` doesn't make sense. You can see in the standard [SWAP note script](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/note_scripts/SWAP.masm) how `RECIPIENT` is used. Here, using a single hash, is sufficient to ensure that the swapped asset and its note can only be consumed by the defined target.
+
+### Note nullifier to ensure private consumption
+The note's nullifier is computed as `hash(serial_num, script_hash, input_hash, vault_hash)`.
 
 This achieves the following properties:
 - Every note can be reduced to a single unique nullifier.
-- We cannot derive a note's hash from its nullifier.
-- To compute the nullifier, we must know all components of the note: `serial_num`, `script_hash`, `input_hash`, and `vault_hash`.
+- One cannot derive a note's hash from its nullifier.
+- To compute the nullifier, one must know all components of the note: `serial_num`, `script_hash`, `input_hash`, and `vault_hash`.
 
 To know a note’s nullifier, one needs to know all details of the note, e.g. the note's serial number. That means if a note is private and the operator stores only the note's hash, only those with the note details know if this note has been consumed already. Zcash first introduced this approach.
```

Comment:
Remove the first sentence it's a duplicate of the 3rd point. 

Link to where Zcash introduced this idea if possible ( blog post or something )